src/wallet/wallet.cpp

```diff
@@ -2125,20 +2125,16 @@ void CWallet::ReacceptWalletTransactions(interfaces::Chain::Lock& locked_chain)
     for (const std::pair<const int64_t, CWalletTx*>& item : mapSorted) {
         CWalletTx& wtx = *(item.second);
         std::string unused_err_string;
-        wtx.SubmitMemoryPoolAndRelay(unused_err_string, false, locked_chain);
+        wtx.SubmitMemoryPoolAndRelay(unused_err_string, false);
     }
 }
 
-bool CWalletTx::SubmitMemoryPoolAndRelay(std::string& err_string, bool relay, interfaces::Chain::Lock& locked_chain)
+bool CWalletTx::SubmitMemoryPoolAndRelay(std::string& err_string, bool relay)
 {
     // Can't relay if wallet is not broadcasting
     if (!pwallet->GetBroadcastTransactions()) return false;
-    // Don't relay coinbase transactions outside blocks
-    if (IsCoinBase()) return false;
```

Comment:
You mean compare to the one at beginning of ATMP ?

Reply:
`BroadcastTransaction` does a convenience check that is only used in the sendrawtransaction (to notify a user of a transaction that is already confirmed). This is done by looking up if at least one of the outputs is in the utxo set. For the purpose of the wallet, this check is completely useless. Previously we wouldn't call it and now we do (even for coinbase txs), where it can be trivially avoided.

Reply:
Ah yes, should the wallet try to check as much as confirmation rules it can, like the coinbase one ? It's a layer violation, but that's trivial as you said.

Reply:
Why is it a layer violation of the wallet to call a [pure method](https://en.wikipedia.org/wiki/Pure_function) on one of it's txs?

Reply:
Really IMO, it's a layer violation because the wallet is doing the job of the node to check consensus rules instead of relying on what the node says.

Reply:
Do you want I update to get back the check ? Wallet are not going to broadcast thousands of coinbase transactions to this be real performance issue.

Reply:
I agree with Marco that this isn't a layer violation. `IsCoinBase()` is a pure utility function and can be called anywhere.

But I also agree with ariard that this isn't a big deal. Calling `IsCoinBase()` here is an optimization that affects a tiny number of transactions for a tiny number of users.

Reply:
Reverted in #16557.